File: 225-implement-stack-using-queues/implement-stack-using-queues.java
class MyStack {

    Queue<Integer> queue1;
    Queue<Integer> queue2;

    public MyStack() {
        queue1 = new ArrayDeque<>();
        queue2 = new ArrayDeque<>();
    }
    
    public void push(int x) {
        
        Queue<Integer> filled_queue = queue1.isEmpty() ? queue2 : queue1;

        filled_queue.add(x);

    }
    
    public int pop() {
        Queue<Integer> filled_queue = queue1.isEmpty() ? queue2 : queue1;
        Queue<Integer> empty_queue = queue1.isEmpty() ? queue1 : queue2;
        
        int size = filled_queue.size();

        for(int i=0 ; i<size-1 ; i++){

            empty_queue.add(filled_queue.poll());

        }
        
        return filled_queue.poll();

    }
    
    public int top() {
        
        Queue<Integer> filled_queue = queue1.isEmpty() ? queue2 : queue1; 
        Queue<Integer> empty_queue = queue1.isEmpty() ? queue1 : queue2;

        int size = filled_queue.size();

        for(int i=0 ; i<size-1 ; i++){
            empty_queue.add(filled_queue.poll());
        }

        int top_elm = filled_queue.poll();
        empty_queue.add(top_elm);
        return top_elm;

    }
    
    public boolean empty() {

        return (queue1.isEmpty() && queue2.isEmpty());

    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */

File: 225-implement-stack-using-queues/implement-stack-using-queues.py
from collections import deque

class MyStack:

    queue1 = None
    queue2 = None

    def __init__(self):
        self.queue1 = deque()
        self.queue2 = deque()

    def push(self, x: int) -> None:
        filled_queue = self.queue1 if self.queue1 else self.queue2
        filled_queue.append(x)

    def pop(self) -> int:
        filled_queue = self.queue1 if self.queue1 else self.queue2
        empty_queue = self.queue1 if not self.queue1 else self.queue2
        
        size = len(filled_queue)
        for _ in range(size-1):
            empty_queue.append(filled_queue.popleft())
        elm = filled_queue[0]
        filled_queue.popleft()
        return elm

    def top(self) -> int:
        
        filled_queue = self.queue1 if self.queue1 else self.queue2
        empty_queue = self.queue1 if not self.queue1 else self.queue2

        size = len(filled_queue)

        for _ in range(size-1):
            empty_queue.append(filled_queue.popleft())
        top_elm = filled_queue.popleft()
        empty_queue.append(top_elm)
        return top_elm

    def empty(self) -> bool:

        return (not self.queue1) and (not self.queue2)        


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()